### ISSUE TITLE
Invalid reads in `ellblock` operations via Fortran API

### ISSUE THREAD

**Issue by nicolasbock** [lanl/bml#342]
Running the Fortran tests through valgrind produces several "invalid read" errors, indicating out of bounds access to arrays (see for example https://travis-ci.org/lanl/bml/jobs/648895583#L4495).

**Comment by nicolasbock** [comment]
@jeanlucf22 I have looked at this a little but haven't found the problem yet. Could you have a look too please? Since this error occurs only with the Fortran API, maybe it's an issue with array base `1` versus `0`? If that's the case, then the issue should be fixable in the Fortran API itself.

**Comment by nicolasbock** [comment]
The calculating of the dimension of `bsize` does not seem to be correct @jeanlucf22 . When I run
```
ctest -R fortran-add-ellblock-single_real-valgrind --verbose
```

I get
```
875: ==1101346== Invalid read of size 4
875: ==1101346==    at 0x1E7F2A: bml_random_matrix_ellblock_single_real (C-interface/ellblock/bml_allocate_ellblock_typed.c:212)
875: ==1101346==    by 0x19F162: bml_random_matrix_ellblock (C-interface/ellblock/bml_allocate_ellblock.c:345)
875: ==1101346==    by 0x19301D: bml_random_matrix (C-interface/bml_allocate.c:412)
875: ==1101346==    by 0x180F99: __bml_allocate_m_MOD_bml_random_matrix (Fortran-interface/bml_allocate_m.F90:232)
875: ==1101346==    by 0x11CF7D: __add_matrix_single_real_MOD_test_add_matrix_single_real (Fortran-tests/add_matrix_single_real.F90:50)
875: ==1101346==    by 0x1170DE: __add_matrix_MOD_test_add_matrix (Fortran-tests/add_matrix.F90:33)
875: ==1101346==    by 0x1164D2: MAIN__ (Fortran-tests/testf.F90:47)
875: ==1101346==    by 0x116E7B: main (Fortran-tests/testf.F90:10)
```

See #344 for the code I was using.

**Comment by nicolasbock** [comment]
Fixed with #346
